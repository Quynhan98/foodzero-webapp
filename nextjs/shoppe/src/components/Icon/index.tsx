import React, { memo } from 'react'
import { Box, Image, BoxProps } from '@chakra-ui/react'

interface IconProps extends BoxProps {
  srcIcon: string
  alt: string
}

const Icon = ({ srcIcon, alt, ...rest }: IconProps) => {
  return (
    <Box display="flex" alignItems="flex-end" {...rest}>
      <Image src={srcIcon} alt={alt} data-testid="image" />
    </Box>
  )
}

export default memo(Icon)
