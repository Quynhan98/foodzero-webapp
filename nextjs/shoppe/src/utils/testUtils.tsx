import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
