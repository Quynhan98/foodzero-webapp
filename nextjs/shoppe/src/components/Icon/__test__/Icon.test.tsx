import React from 'react'

// component
import Icon from '@components/Icon'

// Utils
import { render } from '@utils/testUtils'

describe('Icon render', () => {
  it('Should show match Icon DOM Snapshot', () => {
    const props = {
      alt: 'facebook',
      srcIcon: '/images/facebook.svg',
    }
    const component = render(<Icon {...props} />)
    expect(component).toMatchSnapshot()
  })
})
