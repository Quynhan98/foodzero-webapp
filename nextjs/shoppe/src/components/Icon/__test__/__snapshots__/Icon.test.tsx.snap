// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render Should show match Icon DOM Snapshot 1`] = `
<div>
  <div
    class="css-1nylpq2"
  >
    <img
      alt="facebook"
      class="chakra-image css-0"
      data-testid="image"
      src="/images/facebook.svg"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
